src/Kernel/Process.class.st

```diff
@@ -324,18 +369,17 @@ Process >> isSuspended [
 
 { #category : #testing }
 Process >> isTerminated [
+
+	"Answer if the receiver is terminated, i.e. if the receiver is not active and 
+	one of the following conditions is met:
+	(1) the receiver is a defunct process (suspendedContext = nil or pc = nil)
+	(2) the receiver is suspended within a method with isTerminated pragma"
+
 	self isActiveProcess ifTrue: [ ^ false ].
 	
-	^ suspendedContext isNil or: [
-		"If the suspendedContext is the bottomContext it is the block in Process>>newProcess.
-		If so, and the pc is greater than the startpc, the block has already sent and returned
-		from value and there is nothing more to do."
-		suspendedContext isBottomContext and: [
-			suspendedContext isDead or: [
-				"The pc of the suspendedContext is set to nil in #terminate explicitly.
-				Leaving this line in for safety."
-  	  			suspendedContext pc > suspendedContext startpc ] ] ]
-
+	^suspendedContext isNil or: [
+		suspendedContext isDead or: [
+			(suspendedContext method hasPragmaNamed: #isTerminated)]]
```

Comment:
Our PRs #8567 and #8845 were not compatible to each other.  And I think it leads to some broken tests here. I would exclude #8845 from this PR for now. 
But the correct merge would be: 
- use #isTerminated from my PR #8567.
- modify #isEndOfProcessTermination to use Jaromir logic based on pragma (hasPragma here)

The reason why I suggest to postpone #8845 is because in my PR I followed the idea that the full stepping over the process should be equivalent to the normal execution which always ends up at the final #suspend message. Therefore the real termination status requires explicit check for the #pc to detect such condition. See #isEndOfProcessTermination method.
Thus a simple test proposed in #8845 is not enough. Stepping the process would end up at the start of the method where <isTerminated> pragma is detected. And it will not be equivalent to the normal execution of a process: stepping will not execute "thisContext terminateTo: nil" part of #endProcess method.

Reply:
> But the correct merge would be:
>
> - use #isTerminated from my PR [Very critical fix] Processor>>#terminateRealActive #8567.
> - modify #isEndOfProcessTermination to use Jaromir logic based on pragma (hasPragma here)

Yes, that's what I'd suggest as well; #8845 is not that important - it doesn't fix bugs; just simplifies the logic.